File: oceanlk-backend/src/main/java/com/oceanlk/backend/model/Notification.java
package com.oceanlk.backend.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "notifications")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Notification {
    @Id
    private String id;
    private String message;
    private String type; // "INFO", "WARNING", "ERROR"
    private boolean isRead;
    private LocalDateTime createdAt;
    private String link; // Optional URL (e.g., "/admin/approvals")
    private String recipientRole; // "ROLE_ADMIN", "ROLE_SUPER_ADMIN"
    private String recipientId; // Optional: target a specific admin

    public Notification(String message, String type, String recipientRole, String link) {
        this.message = message;
        this.type = type;
        this.recipientRole = recipientRole;
        this.link = link;
        this.isRead = false;
        this.createdAt = LocalDateTime.now();
    }
}

File: oceanlk-next/models/Notification.ts
import mongoose, { Schema, Document, Model } from 'mongoose';

export interface INotification extends Document {
    message: string;
    type: 'INFO' | 'WARNING' | 'ERROR';
    isRead: boolean;
    link?: string;
    recipientRole?: 'ROLE_ADMIN' | 'ROLE_SUPER_ADMIN';
    recipientId?: string;
    createdAt: Date;
}

const NotificationSchema = new Schema<INotification>({
    message: { type: String, required: true },
    type: {
        type: String,
        enum: ['INFO', 'WARNING', 'ERROR'],
        default: 'INFO'
    },
    isRead: {
        type: Boolean,
        default: false,
        index: true
    },
    link: String,
    recipientRole: {
        type: String,
        enum: ['ROLE_ADMIN', 'ROLE_SUPER_ADMIN'],
        index: true
    },
    recipientId: {
        type: String,
        index: true
    },
    createdAt: {
        type: Date,
        default: Date.now,
        index: { expires: '90d' } // Automatic cleanup after 90 days
    }
}, {
    collection: 'notifications'
});

const Notification: Model<INotification> = mongoose.models.Notification || mongoose.model<INotification>('Notification', NotificationSchema);

export default Notification;
